js/common/bootstrap/modules/buffer.js

```diff
@@ -949,25 +947,45 @@ global.DEFINE_MODULE('buffer', (function () {
     this.parent.writeDoubleBE(value, this.offset + offset, !!noAssert);
   };
 
-  Buffer.from = function (value, encoding) {
-    if (typeof value === 'number') {
-      throw new TypeError('The "value" argument must not be of type number. Received type number');
+  Buffer.from = function (subject, offsetOrEncoding, length) {
+    if (typeof subject === 'number') {
+      throw new TypeError(`The "value" argument must not be of type number. Received type ${typeof subject}`);
+    }
+    if (Array.isArray(subject)) {
+      return createBufferFromByteArray(subject);
+    }
+    if (typeof subject === 'string') {
+      return createBufferFromString(subject, offsetOrEncoding);
+    }
+    if (subject instanceof ArrayBuffer) {
+      return createBufferFromArrayBuffer(subject, offsetOrEncoding, length);
+    }
+    if (Buffer.isBuffer(subject)) {
+      return createBufferFromBuffer(subject, offsetOrEncoding, length);
+    }
+    if (typeof subject[Symbol.toPrimitive] === 'function') {
+      return Buffer.from(subject[Symbol.toPrimitive](), offsetOrEncoding, length);
+    }
+    if (typeof subject.valueOf === 'function') {
+      return Buffer.from(subject.valueOf(), offsetOrEncoding, length);
     }
-    return new Buffer(value, encoding);
+    throw new TypeError(`The first argument must be one of type string, Buffer, ArrayBuffer, Array or Array-like Object. Received type ${typeof subject}`);
   };
 
   Buffer.of = function (...values) {
-    return new Buffer(values);
+    return Buffer.from(values);
   };
 
   Buffer.alloc = function (size) {
-    const buf = new Buffer(size);
-    buf.alloc(size);
-    return buf;
+    return Buffer.from(Array(size).fill(0));
   };
 
   Buffer.allocUnsafe = Buffer.allocUnsafeSlow = function (size) {
-    return new Buffer(size);
+    const buffer = Object.create(Buffer.prototype);
+    buffer.length = size;
+    buffer.offset = 0;
+    buffer.parent = new SlowBuffer(size);
+    return new Proxy(buffer, BUFFER_PROXY_HANDLER);
```

Comment:
This method is still used by some libraries because it's slightly faster than zero-filling the buffer. Note that `new Buffer(size)` zero-fills in newer Node versions for security reasons and we now mimic that behavior.